src/cplscheme/BaseCouplingScheme.cpp

```diff
@@ -148,6 +148,12 @@ void BaseCouplingScheme::sendData(const m2n::PtrM2N &m2n, const DataMap &sendDat
         m2n->send(serialized.gradients(), data->getMeshID(), data->getDimensions() * data->meshDimensions() * serialized.nTimeSteps());
       }
     } else {
+      if (initialCommunication) { // send data for WINDOW_START. Will be used for WINDOW_START and WINDOW_END.
+        PRECICE_ASSERT(math::equals(stamples.front().timestamp, time::Storage::WINDOW_START));
+        PRECICE_ASSERT(math::equals(stamples.back().timestamp, time::Storage::WINDOW_START));
+      } else { // send data for WINDOW_END.
+        PRECICE_ASSERT(math::equals(stamples.back().timestamp, time::Storage::WINDOW_END));
+      }
```

Comment:
Currently, `initialCommunication` only triggers some assertions in `sendData`. Looks suspicious, no?
Or better remove the argument?

Reply:
I think we can keep it. The assertions are generally a good safeguard in my opinion and we might add more functionality later, if we don't want to send/receive values for `WINDOW_START` to save some computational cost.